projects/extension/ai/load_dataset.py

```diff
@@ -58,30 +66,57 @@ def create_table(
         # Replace any non-alphanumeric characters with underscore
         table_name = "".join(c if c.isalnum() else "_" for c in base_name.lower())
 
-    # Construct fully qualified table name
-    qualified_table = f'"{schema}"."{table_name}"'
+        # Check table name length - PostgreSQL has a 63 character limit for identifiers
+        if len(table_name) > 63:
+            # Find the last underscore before the 63 character limit
+            last_underscore = table_name[:63].rstrip("_").rfind("_")
+            if last_underscore > 0:
+                table_name = table_name[:last_underscore]
+            else:
+                # If no underscore found, just truncate
+                table_name = table_name[:63]
+    else:
+        # table_name is provided by the user
+        # Check table name length - PostgreSQL has a 63 character limit for identifiers
+        if len(table_name) > 63:
+            plpy.error(
+                f"Table name '{table_name}' exceeds PostgreSQL's 63 character limit"
+            )
 
-    # Check if table exists
+    # Construct fully qualified table name
     plan = plpy.prepare(
         """
-        SELECT pg_catalog.to_regclass(pg_catalog.format('%I.%I', $1, $2)) is not null as exists
+        SELECT pg_catalog.format('%I.%I', $1, $2) as qualified_table_name
     """,
         ["text", "text"],
     )
     result = plan.execute([schema, table_name], 1)
-    table_exists = result[0]["exists"]
+    qualified_table = result[0]["qualified_table_name"]
+
+    # Check if table exists
+    result = plpy.execute(
+        f"""
+        SELECT pg_catalog.to_regclass('{qualified_table}')::text as friendly_table_name
```

Comment:
does this strip the schema if the search_path allows for it?

Reply:
yes it does. Is that a problem?

Reply:
No. Just curious